pkg/gofr/datasource/cassandra/cassandra.go

```diff
@@ -125,20 +130,20 @@ func (c *Client) Query(dest interface{}, stmt string, values ...interface{}) err
 	default:
 		c.logger.Debugf("a pointer to %v was not expected.", rv.Kind().String())
 
-		return unexpectedPointer{target: rv.Kind().String()}
+		return ErrUnexpectedPointer{target: rv.Kind().String()}
```

Comment:
Now, the code is clearer to me. Thanks for the changes by the way.

I would like to suggest something that would make more sense according to me

Here the code relies (and already relied) on something that single purpose could be defined by these:
- we need to be able to catch this error
- we want to provide a meaningful error message that will contain the kind.

The current way is OK, but somehow overkill

Here is what I would code if I had to.

We don't define a struct, we define a simple sentinel error. This will ensure we can catch the error with a simple `errors.Is`

```go
var ErrUnexpectedPointer = errors.New("unexpected pointer")
```

Then we use `fmt.Errorf("%w to %s", ErrUnexpectedPointer, rv.Kind.String())` here

I know code was there, I know I'm asking to change something you are not responsible for. But, I report what I would have expected here.

As the errors are now exported, I would expect them to be clean.

Please let me know what you think about it

Reply:
I think so, the other changes that could also needed is to rewrite the possible existing errors.As using this struct to an errors.Is

Anyway, I will let maintainers tell us what they thi k about my suggestion

@vipul-rawat @srijan-27 @Umang01-hash maybe?

Reply:
Yes @ccoVeille - I agree with you, if the errors are now being exported then it needs to be handled neatly.
But I think, we can separate out these changes from this PR. Let's create a different issue for that, and follow up there.

Reply:
Of course, I raise my points, but I didn't expect all to be fixed in this PR

Could you open an issue about it?

Reply:
So maybe for now, in this PR we could avoid to change error variable to exported. This way they won't be available, then changed later, leading to breaking changes

Reply:
@ssshekhu53 - Please revert the changes as required. Will take these up in separate PR.

Reply:
For records #893 has been opened